__tests__/api.test.js

```diff
@@ -150,4 +151,60 @@ describe('GET /api/articles/:article_id/comments', () => {
                 expect(body.msg).toBe('Bad request');
             });
     });
+});
+
+describe('POST /api/articles/:article_id/comments', () => {
+    test("201: responds with the posted comment and adds the comment to an article", () => {
+        const newComment = {
+            username: 'butter_bridge',
+            body: 'this article sucks'
+        };
+        return request(app)
+        .post("/api/articles/1/comments")
+        .send(newComment)
+        .expect(201)
+        .then(({body}) => {
+            expect(body.comment).toMatchObject({
+                author: 'butter_bridge',
+                body: 'this article sucks'
+            })
+        })
+    })
+    test("400: responds with an appropriate status and error message when provided with a non-existent user-name", () => {
+        const newComment = {
+            body: 'this article sucks'
+        };
+        return request(app)
+        .post("/api/articles/1/comments")
+        .send(newComment)
+        .expect(400)
+        .then(({body}) => {
+            expect(body.msg).toBe('Username does not exist')
```

Comment:
It's correct that this is a 400 as the newComment is missing a username, but the message should reflect this, rather than saying that the username doesn't exist. We should however have a separate test here for a 404 where we provide a username that doesn't exist 👍